Add `getproject` command

yggi49 opened the pull request:
Use `pew getproject [env]` to print the given virtualenv's project directory, if one has been set.  If the `env` parameter is omitted, use the currently active virtualenv.

yggi49 commented:
Note on the rationale: coming from virtualenvwrapper, I am used to call `cdproject` to switch back to the current virtualenv’s project root directory.  With `pew getproject` I am able to define a `cdproject` shell function that accomplishes the same feat.

meshy commented:
I'm not sure if you're using `pipenv`, but it may be useful for you to know that you can achieve something similar with `pipenv --where`.

yggi49 commented:
No, I am not using `pipenv`.

That said, I think that if `pew` offers a `sitepackages_dir`, then offering a corresponding command to get the project directory makes sense as well.

ashwinvis commented:
@meshy I use pipenv, yet I think this PR is handy. For the following reasons:

1. `pipenv --where` will only work when it can find a Pipfile in the current directory or parent directories. If not,

```
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project.
No Pipfile present at project home. Consider running `pipenv install` first to automatically generate a Pipfile for you.
```
2. For some reason, I cannot filter out pipenv's `Courtesy notice: Pipenv found itself running within a virtual environment ...`, despite using something like:

```
~/s/s/pew ❯❯❯ pipenv --where | tail -n 1                                                                                                                                                   (sandbox) 
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project.
/home/avmo/src/sandbox/pew
```

meshy commented:
@ashwinvis I hadn't considered that use case -- seems like a reasonable one!

As for the `Courtesy notice`, I suspect that's going to `stderr`, not `stdout`, so the pipe (`|`) isn't redirecting that.

yggi49 commented:
After a year of waiting: do you think there is any chance of getting something done in this direction? =)  Are you still missing something that you would like to have added prior to considering a merge?

pfmoore commented:
Hi - thanks for your patience. Real life issues have got in the way of progressing a lot of things recently, but we're starting to pick up on the backlog now. This looks good to me - there are a few things I'd like to do (use argparse, so we get proper help for the subcommand, and adding a test, for example) but I think that we can defer those to a follow-up PR, rather than holding this up any longer.

I'd like to do a quick review before merging this - I don't have the time to do so right now, but I'll sort it out tomorrow.

yggi49 commented:
> there are a few things I'd like to do (use argparse, so we get proper help for the subcommand, and adding a test, for example)

I just switched to using argparse and added a small test suite.  Please let me know if there is something else that you are missing.

pfmoore commented:
Thanks for the contribution!

yggi49 commented:
My pleasure—thank you for merging!
